gammapy/analysis/core.py

```diff
@@ -246,11 +246,14 @@ def _map_making(self):
                 geom_settings.axes.energy_true
             )
         geom_irf["binsz_irf"] = geom_settings.wcs.binsize_irf.to("deg").value
-        offset_max = geom_settings.selection.offset_max
         log.info("Creating datasets.")
 
         maker = MapDatasetMaker(selection=self.config.datasets.map_selection)
-        maker_safe_mask = SafeMaskMaker(methods=["offset-max"], offset_max=offset_max)
+
+        safe_mask_selection = self.config.datasets.safe_mask.methods
+        safe_mask_settings = self.config.datasets.safe_mask.settings.dict()
+        offset_max = safe_mask_settings['offset_max']
+        maker_safe_mask = SafeMaskMaker(methods=safe_mask_selection, **safe_mask_settings)
```

Comment:
Does `maker_safe_mask` fails when is run without a method (i.e. an empty list)? 
In that case we should prevent it from running it.

Reply:
Normally it does not. I assume that the empty list would mean no safe mask applied.